### PR TITLE
updated to support directive on type

### PR DESCRIPTION
**Pull request by dragonfriend0013**
Added code to wrap all fields with auth when an @auth directive is present on a type.  added 3 new test cases

### CODE REVIEW

**Comment by dragonfriend0013** [comment]
Closes #24

**Comment by jonnydgreen** [comment]
Nice, I'll have a look tonight! :)

**Review by mcollina — APPROVED** [review]
lgtm

**Review by jonnydgreen — APPROVED** [review]
LGTM - nice!

**Review by jonnydgreen — COMMENTED** [review]
Functionally looks great, I've just had a thought - could you add an example (see the examples folder) and add a section in this docs [file](https://github.com/mercurius-js/auth/blob/main/docs/auth-directive.md) showing we now support at the type level as well?

**Comment by dragonfriend0013** [comment]
Yes, I can do that

**Comment by dragonfriend0013** [comment]
i have updated the docs and a new example.  Let me know if you see anything else.

File: examples/gateway2.js [hunk 1]
```diff
@@ -0,0 +1,192 @@
+'use strict'
```
Comment (jonnydgreen) [inline examples/gateway2.js:1]: Looks great, thanks for doing that! Could we make the file name more meaningful, such as `gateway-with-auth-on-type.js` for example?

**Review by jonnydgreen — APPROVED** [review]
LGTM - thanks very much for your work on this! I'll release this tonight

**Pull request merged**